api/nativeMessaging/main.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+let port = null;
+
+const appendMessage = (text) => {
+  document.getElementById('response').innerHTML += '<p>' + text + '</p>';
```

Comment:
we want to discourage `.innerHTML`.

```js
(text, data) => 
  const p = document.createElement('p');
  p.append(text);

  // to show a result alongside text
  if (data) {
    const strong = document.createElement('strong');
    strong.append(data);
    p.append(strong);
  }

  document.getElementById('response').append(p);
}
```

Reply:
I did minimal updating to existing JavaScript. I normally use `.textContent`. Should not be critical to the extension, can update.

Reply:
Is there an existing demo? I asked that at the top-level. Some of my review is moot if we're just porting old code, although it's nice to improve it along the way.

Reply:
I am not sure what you mean by "existing demo"? I linked to the MV2 example in this repository at https://github.com/GoogleChrome/chrome-extensions-samples/pull/617#issuecomment-903443805.

Reply:
@samthor I resolved some review points you noted. I am not sure where you are getting `data` from?